src/Raven.Server/ServerWide/Commands/Sharding/UpdatePrefixedShardingSettingCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Sharding;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide.Sharding;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands.Sharding
+{
+    public sealed class UpdatePrefixedShardingSettingCommand : UpdateDatabaseCommand
+    {
+        public PrefixedShardingSetting Setting;
+
+        public UpdatePrefixedShardingSettingCommand()
+        {
+        }
+
+        public UpdatePrefixedShardingSettingCommand(PrefixedShardingSetting setting, string database, string raftId) : base(database, raftId)
+        {
+            Setting = setting;
+        }
+
+        public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
+        {
+            var location = record.Sharding.Prefixed.BinarySearch(Setting, PrefixedSettingComparer.Instance);
```

Comment:
can you call `UpdatePrefixedShardingSettingCommand` without having a prior `Prefixed` set? this will cause NRE..